tool/tsh/config.go

```diff
@@ -105,13 +121,29 @@ func onConfig(cf *CLIConf) error {
 	fmt.Fprintln(&sb)
 	fmt.Fprintf(&sb, "#\n# Begin generated Teleport configuration for %s from `tsh config`\n#\n", tc.Config.WebProxyAddr)
 
-	err = writeSSHConfig(&sb, rootClusterName, knownHostsPath, proxyHost, proxyPort, false)
+	err = writeSSHConfig(&sb, hostConfigParameters{
+		ClusterName:         rootClusterName,
+		KnownHostsPath:      knownHostsPath,
+		IdentityFilePath:    identityFilePath,
+		CertificateFilePath: keypaths.SSHCertPath(keysDir, proxyHost, tc.Config.Username, rootClusterName),
+		ProxyHost:           proxyHost,
+		ProxyPort:           proxyPort,
+		TSHPath:             cf.executablePath,
+	})
 	if err != nil {
 		return trace.Wrap(err)
 	}
 
 	for _, leafCluster := range leafClusters {
-		err = writeSSHConfig(&sb, leafCluster.GetName(), knownHostsPath, proxyHost, proxyPort, true)
+		err = writeSSHConfig(&sb, hostConfigParameters{
+			ClusterName:         leafCluster.GetName(),
+			KnownHostsPath:      knownHostsPath,
+			IdentityFilePath:    identityFilePath,
+			CertificateFilePath: keypaths.SSHCertPath(keysDir, proxyHost, tc.Config.Username, rootClusterName),
```

Comment:
should this use `leafCluster.GetName()` instead of `rootClusterName`?

Reply:
I believe the root cluster's is the only cert we can guarantee will exist, tsh doesn't cache certs for leaf clusters unless specifically asked. The leaf clusters trust should all trust the root cert anyway, so it should work fine.